src/services/codefixes/convertToAsyncFunction.ts

```diff
@@ -338,44 +347,54 @@ namespace ts.codefix {
         return varDeclList ? [varDeclList, tryStatement] : [tryStatement];
     }
 
+    function getIdentifierTextsFromBindingName(bindingName: BindingName): ReadonlyArray<string> {
+        if (isIdentifier(bindingName)) return [bindingName.text];
+        return flatMap(bindingName.elements, element => {
+            if (isOmittedExpression(element)) return [];
+            return getIdentifierTextsFromBindingName(element.name);
+        });
+    }
+
     function createUniqueSynthName(prevArgName: SynthIdentifier) {
         const renamedPrevArg = createOptimisticUniqueName(prevArgName.identifier.text);
         const newSynthName = { identifier: renamedPrevArg, types: [], numberOfAssignmentsOriginal: 0 };
         return newSynthName;
     }
 
-    function transformThen(node: CallExpression, transformer: Transformer, outermostParent: CallExpression, prevArgName?: SynthIdentifier): ReadonlyArray<Statement> {
+    function transformThen(node: CallExpression, transformer: Transformer, outermostParent: CallExpression, prevArgName?: SynthBindingName): ReadonlyArray<Statement> {
         const [res, rej] = node.arguments;
 
         if (!res) {
             return transformExpression(node.expression, transformer, outermostParent);
         }
 
-        const argNameRes = getArgName(res, transformer);
+        const argNameRes = getArgBindingName(res, transformer);
         const transformationBody = getTransformationBody(res, prevArgName, argNameRes, node, transformer);
 
         if (rej) {
-            const argNameRej = getArgName(rej, transformer);
+            const argNameRej = getArgBindingName(rej, transformer);
 
             const tryBlock = createBlock(transformExpression(node.expression, transformer, node, argNameRes).concat(transformationBody));
 
             const transformationBody2 = getTransformationBody(rej, prevArgName, argNameRej, node, transformer);
 
-            const catchArg = argNameRej ? argNameRej.identifier.text : "e";
-            const catchClause = createCatchClause(catchArg, createBlock(transformationBody2));
+            const catchArg = argNameRej ? "identifier" in argNameRej ? argNameRej.identifier.text : argNameRej.bindingPattern : "e";

```

Comment:
seems like this could be a helper function than checking this ( I noticed at least two places)

Reply:
Done! Meant to clean that up before pushing.